src/Workspaces/CSharp/Portable/CodeGeneration/PropertyGenerator.cs

```diff
@@ -54,7 +54,7 @@ public static bool CanBeGenerated(IPropertySymbol property)
             IList<bool> availableIndices)
         {
             var declaration = GeneratePropertyOrIndexer(property, GetDestination(destination),
-                options, destination?.SyntaxTree.Options ?? options.ParseOptions);
+                options, options.ParseOptions ?? destination?.SyntaxTree.Options);
```

Comment:
Hi @CyrusNajmabadi
I looked into the commit history and found you change this code in this commit 7ce9f4e3c0a3970e2a0ddc02058c61632c543b7e  
(together a lot of other same changes) 
And the linked PR https://github.com/dotnet/roslyn/pull/14540 doesn't have any linked issue.

Do you feel this change 
``` destination?.SyntaxTree.Options ?? options.ParseOptions``` to
```options.ParseOptions ?? destination?.SyntaxTree.Options ```
 would cause any problem?

I am doing this because the linking bug for this PR is caused in this loop,
http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/CodeGeneration/AbstractCodeGenerationService.cs,278

When there are multiple symbols passed into the GenerationService, every time a new destination syntax node is generated, its parse option is C# 9. So if here we prefer the options of syntax tree, at the first time it is called, it would generate C# 5 compatible property, and the next property would be C# 9 compatible property.